Danger PR Review Missing Permissions

On PRs coming from forks, contributors do not have write permissions to add a comment via the Danger GitHub Action.

Example: https://github.com/Comfy-Org/ComfyUI_frontend/actions/runs/16259117396/job/45922839976
```bash
Run npx danger ci
  npx danger ci
  shell: /usr/bin/bash -e {0}
  env:
    GITHUB_TOKEN: ***
    DANGER_GITHUB_API_TOKEN: ***

Failing the build, there is 1 fail.
Request failed [403]: https://api.github.com/repos/Comfy-Org/ComfyUI_frontend/issues/4361/comments
Response: {
  "message": "Resource not accessible by integration",
  "documentation_url": "https://docs.github.com/rest/issues/comments#create-an-issue-comment",
  "status": "403"
}
Feedback: undefined
Could not add a commit status, the GitHub token for Danger does not have access rights.
If the build fails, then danger will use a failing exit code.
Error: Process completed with exit code 1.
```

Related docs: https://docs.github.com/en/actions/how-tos/writing-workflows/choosing-what-your-workflow-does/controlling-permissions-for-github_token

┆Issue is synchronized with this [Notion page](https://www.notion.so/Issue-4448-Danger-PR-Review-Missing-Permissions-2306d73d365081b8bfb0f0210f2a4c46) by [Unito](https://www.unito.io)

--- a/.github/workflows/danger.yaml
+++ b/.github/workflows/danger.yaml
@@ -23,5 +23,4 @@ jobs:
       - name: Run Danger
         run: npx danger ci
         env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           DANGER_GITHUB_API_TOKEN: ${{ secrets.GITHUB_TOKEN }}

--- a/.github/workflows/danger.yaml
+++ b/.github/workflows/danger.yaml
@@ -1,7 +1,13 @@
 name: Danger PR Review
 on:
-  pull_request:
+  pull_request_target:
     types: [opened, edited, synchronize]
+
+permissions:
+  contents: read
+  issues: write # What Danger needs to comment on PRs
+  pull-requests: write
+  statuses: write
 
 jobs:
   danger:
@@ -10,6 +16,7 @@ jobs:
       - name: Checkout code
         uses: actions/checkout@v4
         with:
+          ref: ${{ github.event.pull_request.head.sha }}
           fetch-depth: 0
 
       - name: Setup Node.js

--- a/.github/workflows/danger.yaml
+++ b/.github/workflows/danger.yaml
@@ -30,4 +30,5 @@ jobs:
       - name: Run Danger
         run: npx danger ci
         env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           DANGER_GITHUB_API_TOKEN: ${{ secrets.GITHUB_TOKEN }}

